larreco/TrackFinder/PMAlgTrackMaker_module.cc

```diff
@@ -0,0 +1,1368 @@
+  std::map< unsigned int, std::vector<size_t> > tried_clusters;
+
+  // temporary set of possible solutions of the selected cluster and clusters in complementary views
+  pma::TrkCandidateColl fCandidates;
+
+  int maxCluster(
+    const std::vector< art::Ptr<recob::Cluster> >& clusters,
+    size_t min_clu_size,
+    geo::View_t view, unsigned int tpc, unsigned int cryo);
+  int maxCluster(
+	int first_idx_tag,
+    const std::vector< art::Ptr<recob::Cluster> >& clusters,
+    float tmin, float tmax, size_t min_clu_size,
+    geo::View_t view, unsigned int tpc, unsigned int cryo);
+
+  void freezeBranchingNodes(pma::TrkCandidateColl & tracks);
+  void releaseAllNodes(pma::TrkCandidateColl & tracks);
+
+  bool areCoLinear(
+	pma::Track3D* trk1, pma::Track3D* trk2,
+	double& dist, double& cos, bool& reverseOrder,
+	double distThr, double distThrMin,
+	double distProjThr,
+	double cosThr);
+  bool mergeCoLinear(pma::TrkCandidateColl & tracks);
+  void mergeCoLinear(tpc_track_map& tracks);
+
+  bool areCoLinear(
+		double& cos3d,
+		TVector3 f0, TVector3 b0, TVector3 f1, TVector3 b1,
+		double distProjThr);
+  void matchCoLinearAnyT0(pma::TrkCandidateColl & tracks);
+
+  bool reassignHits(
+	const std::vector< art::Ptr<recob::Hit> >& hits,
+	const std::vector< art::Ptr<recob::Cluster> > & clusters,
+	pma::TrkCandidateColl & tracks, size_t trk_idx, double dist2);
+  bool reassignHits(
+	const std::vector< art::Ptr<recob::Hit> >& hits,
+	pma::TrkCandidateColl & tracks, size_t trk_idx, double dist2);
+
+  double collectSingleViewFront(pma::Track3D & trk,
+	std::vector< art::Ptr<recob::Hit> > & hits);
+  double collectSingleViewEnd(pma::Track3D & trk,
+	std::vector< art::Ptr<recob::Hit> > & hits);
+
+  bool reassignSingleViewEnds(pma::TrkCandidateColl & tracks,
+	const std::vector< art::Ptr<recob::Cluster> > & clusters);
+  bool reassignSingleViewEnds(pma::TrkCandidateColl & tracks);
+  void guideEndpoints(pma::TrkCandidateColl & tracks);
+
+  double validate(pma::Track3D& trk, unsigned int testView);
+  recob::Track convertFrom(const pma::Track3D& src);
+  // ------------------------------------------------------
+
+  art::ServiceHandle< geo::Geometry > fGeom;
+  const detinfo::DetectorProperties* fDetProp;
+
+  // ******************* tree output **********************
+  int fEvNumber;        // event number
+  int fTrkIndex;        // track index in the event
+  int fPidTag;          // Tag: 0=trk-like, 1=cascade-like
+  double fLength;       // track length
+  double fHitsMse;      // MSE of hits: mean dist^2 of hit to 2D track projection
+  double fSegAngMean;   // Mean segment-segment 3D angle.
+  TTree* fTree_trk;     // overall info
+
+  // ******************** fcl parameters **********************
+  std::string fHitModuleLabel; // label for hits collection (used for trk validation)
+  std::string fCluModuleLabel; // label for input cluster collection
+  int fCluMatchingAlg;         // which algorithm for cluster association
```

Comment:
Consider defining an enum to enumerate all valid values.  Make the data member of enum type.